include/ensmallen_bits/nsga2/nsga2_impl.hpp

```diff
@@ -0,0 +1,404 @@
+/**
+ * @file nsga2_impl.hpp
+ * @author Sayan Goswami
+ *
+ * Implementation of the NSGA-II algorithm. Used for multi-objective
+ * optimisation problems on arbitrary functions.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more Information.
+ */
+
+
+#ifndef ENSMALLEN_NSGA2_NSGA2_IMPL_HPP
+#define ENSMALLEN_NSGA2_NSGA2_IMPL_HPP
+
+#include "nsga2.hpp"
+#include <set>
+#include <vector>
+#include <map>
```

Comment:
I don't have any opposition to adding these to `ensmallen.hpp`; what do you think?

Reply:
Yeah, I will move them to `ensmallen.hpp`. :)